src/mon/HealthMonitor.cc

```diff
@@ -682,6 +683,31 @@ bool HealthMonitor::check_leader_health()
 
   health_check_map_t next;
 
+ // DAEMON_VERSION_INCORRECT
+  std::map<int, std::pair<int, string>> wrong_version;
+  double timeout = 600;
+  if (mon->check_daemon_versions(wrong_version)){
+    time_t current = time(NULL);
+    if ((timeout < difftime(current, start))){
+      ostringstream ss, ds;
+      if (wrong_version.size() == 1){
```

Comment:
My suggested map (see below in _check_daemon_version()_) you would know that if map.size() > 1 then there are more than 1 version running.  From there you'd have to sort out what to say about each version and its list of daemons.  For example, if there are only 2 versions then the higher version is the **correct** one and you could list the daemons with the lower version.  The map could also sort the versions low to high.  So you know that the last entry in the map has the target version of the cluster.

Reply:
I see so then simply create a map inside check_daemon_version() where the key is the ceph version and compare the version of the current daemon to see if it's a key in the map if not add it to the map and add the daemon id to that key therefore still allowing the program to keep track of which daemon is running which version.